### ISSUE TITLE
contrib/google.golang.org/grpc/client Ignored Methods option for Unary Client Interceptor is not Respected

### ISSUE THREAD

**Issue by mohamedpsc** [DataDog/dd-trace-go#1491]
I am trying to ignore creating spans on the grpc client using within the grpc DialOption as follows
```
[]grpc.DialOption{
    // some options
    grpc.WithChainUnaryInterceptor(
          grpctrace.WithIgnoredMethods(ignoredMethods...),
    )
)
```
However these methods are not being ignored. it is working fine on the server side although I am using the same values for the ignoredMethods.

I was using `gopkg.in/DataDog/dd-trace-go.v1 v1.39.0` when I was having this issue
I also updated to `gopkg.in/DataDog/dd-trace-go.v1 v1.42.1` but the issue still persist

References:
https://github.com/DataDog/dd-trace-go/blob/31366ddbe8689fcb73f1874e731668a2360c6a9d/contrib/google.golang.org/grpc/client.go#L168

I believe there should be a check if the method should be ignored before starting the span on the line above

**Comment by mohamedpsc** [comment]
We do have a couple of grpc methods that we don't want to trace and we are using this option on server side to ignore them however we are not able to do so on the client side so we end up having traces for them on Datadog for just the client call

**Comment by ajgajg1134** [comment]
Hello! We're currently evaluating some options for how to approach this:
1. Change `WithIgnoredMethods` to also ignore client-side spans
    Almost certainly won't do this one since it would break backwards compatibility
2. Add a new thing `WithClientIgnoredMethods` that only works for client-side ignoring
3. Wait, release a single option that works for server and client in v2
4. Mark `WithIgnoredMethods` as deprecated, add a `WithUntracedMethods` that works for both

We are currently leaning towards option #4. If this issue is currently blocking you feel free to open an issue with Datadog Support to escalate this feature request (and possibly provide a workaround in the interim).

**Comment by mohamedpsc** [comment]
@ajgajg1134 any ETA for releasing this change? I still can't see it included in any the latest release. am I missing anything?
